controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -288,9 +268,10 @@ public Application createApplication(ApplicationId id, Optional<NToken> token) {
     public ActivateResult deployApplication(ApplicationId applicationId, Zone zone,
                                             ApplicationPackage applicationPackage, DeployOptions options) {
         try (Lock lock = lock(applicationId)) {
-            LockedApplication application = get(applicationId, lock).orElse(new LockedApplication(
+            // TODO: Shouldn't this go through the above method? Seems you can cheat the checks here ... ?
+            LockedApplication application = get(applicationId).map(application1 -> new LockedApplication(application1, lock)).orElse(new LockedApplication(
```

Comment:
Will we break anything if this is tightened?

Reply:
Any comment on this, though?

Reply:
It seems you can create an application with a non-existent tenant here, by shortcutting createApplication. 
This method is called with an application id with a tenant name parsed directly from the URI. 
You won't be able to deploy this to prod, but you'll get an invalid application entry, and one which you can deploy to dev, etc..

Reply:
Ok, there's probably no particular good reason it's like this ...